block/deserialize.go

```diff
@@ -0,0 +1,341 @@
+package block
+
+import (
+	"bytes"
+	"errors"
+
+	"github.com/vulpemventures/go-elements/internal/bufferutil"
+
+	"github.com/vulpemventures/go-elements/transaction"
+)
+
+func deserialize(buf *bytes.Buffer) (*Block, error) {
+	header, err := DeserializeHeader(buf)
+	if err != nil {
+		return nil, err
+	}
+
+	transactions, err := DeserializeTransactions(buf)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Block{
+		Header: header,
+		TransactionsData: &Transactions{
+			Transactions: transactions,
+		},
+	}, nil
+}
+
+func DeserializeHeader(
+	buf *bytes.Buffer,
+) (*Header, error) {
+	d := bufferutil.NewDeserializer(buf)
+
+	version, err := d.ReadUint32()
+	if err != nil {
+		return nil, err
+	}
+
+	isDyna := false
+	if version>>31 == 1 {
+		isDyna = true
+	}
+	version = blockVersion
```

Comment:
can you explain why here the deserialized version is overwritten by the constant one?

Reply:
In [Rust impl](https://github.com/ElementsProject/rust-elements/blob/0d67c57afa1137ab27861bb8c2190413929d4301/src/block.rs#L315) version is overwritten in case of isDyna true but even if it is false value is equal to blockVersion, anyways to be on the same page i will do the same.

Reply:
Done

Reply:
So, I guess the current `blockVersion`value is the result of some block version & 0x7fffffff.. Are we sure liquid blocks have always the same version though? If yes, I recommend adding a comment before this assignment to specify why it's done.
Otherwise, we should better follow rust-elements and make an AND bit-a-bit operation using the same hex value declared there.

What do you think?

Reply:
Lets fellow Rust elements.
Done